### ISSUE TITLE
Ps3 xinput controller with dualshock problems

### ISSUE THREAD

**Issue by NelsonRosenberg** [libretro/RetroArch#3958]
I'm using ScpToolkit to get my PS3 DualShock controller to work as a Xinput controller. Everything runs perfect, except for the rumble. I've tested it outside retroarch and it works fine, but in retroarch I set everything correctly (Joypad w/Analog + Dualshock option in quick menu) and sometimes it works, sometimes it doesn't. I've noticed that when booting up, loading a save state and the option for Dualshock is already pre-selected, it does not work, and I have to select None, go back to the game, go back to the menu and then select DualShock to get rumble. I'm not sure why is it so inconsistent. Can the save state be showing the correct config but i actually configured wrong in some way?

Thank you.

**Comment by lasers** [comment]
I don't know if this will work... This is what I wanted you to try.

My assumption: Everything works works differently between various controllers, hotplugging, hitting "Home" button on the controller to activate/reset, and/or how RetroArch reacts to non-activated controllers (via deadzone?) at startup (which could gimp other inputs like my keyboard)... but can be fixed by activating or unplugging the controller... Also, some people may plug in holding or pushing analog sticks resulting in different settings... and games could be programmed slightly different... so they could react to inputs slightly different too.

Plug in the controller not touching buttons + analog sticks.
Activate the controller by tapping PS button. Two times if you like. Double tap.
Start RetroArch and try. See if that does anything.

The idea is to stay consistent, always activate the controller before you start RetroArch. If you forget to do that, just activate the controller and restart RetroArch. I think if you didn't do that, you hit the PS button later (to return back to the menu?) and it just works afterward.

**Comment by NelsonRosenberg** [comment]
Hi lasers!

Thank you very much for the input, I'll try the things you mentioned and get back to you as soon as I can.

**Comment by lasers** [comment]
How soon? I look forward to hear back from you as soon as possible.

**Comment by NelsonRosenberg** [comment]
Sorry, I took a while to test again. But I think my problem was completely unrelated to the controller itself. I had the config for rumble wrong in retroarch when I started the game and then I created a savestate, restarted retroarch and fixed the rumble config. When I loaded the savestate, the config was wrong again (or corrupted, maybe) and I had to enable and disable rumble several times until it worked. So I think the savestate with the wrong config was causing the problems somehow. Once I deleted that save and started fresh with the correct config, everything is working fine!
